pandas/core/frame.py

```diff
@@ -4621,9 +4624,15 @@ def drop_duplicates(
         if inplace:
             (inds,) = (-duplicated)._ndarray_values.nonzero()
             new_data = self._data.take(inds)
+
+            if ignore_index:
+                new_data = new_data.reset_index(drop=True)
             self._update_inplace(new_data)
         else:
-            return self[-duplicated]
+            result = self[-duplicated]
+            if ignore_index:
+                return result.reset_index(drop=True)
```

Comment:
here you have to copy before you update the index

Reply:
i did not use copy, but `.index` assignment, and i think it is also correct.